armbian/base/scripts/systemd-update-checks.sh

```diff
@@ -37,6 +37,54 @@ redis_set "base:version" "${VERSION}"
 # update onion addresses in Redis
 updateTorOnions
 
+# check for reset triggers on flashdrive
+if FLASHDRIVE="$(/opt/shift/scripts/bbb-cmd.sh flashdrive check)"; then
+    echo "RESET: device ${FLASHDRIVE} detected"
+    if /opt/shift/scripts/bbb-cmd.sh flashdrive mount "${FLASHDRIVE}"; then
+        echo "RESET: flashdrive mounted"
+
+        # are all necessary files for a reset present?
+        if [[ -f /mnt/backup/.reset-token ]] && [[ -f /data/reset-token-hashes ]]; then
+            FLASHDRIVE_TOKEN_HASH="$(echo -n "$(</mnt/backup/.reset-token)" | sha256sum | tr -d "[:space:]-")"
```

Comment:
This seems really unsafe. What if the file contains another expression, for example `$($(ls)\n`? This expression will first be interpreted by the 256sum and could be used to point to the /data/reset-token-hashes files. Maybe run the read string through a sanitizer first?

Reply:
Good input thanks. I can easiyl sanatize the token, as its a SHA256 string with only hex characters allowed.

Example
```console
$ RESET_TOKEN='abcdefghijklmnopqrstuvwxyzABCDXYZ1234567890!@#$%^&*()_+/?|}{}|":' && echo $RESET_TOKEN && echo ${RESET_TOKEN//[^a-f0-9]/}
abcdefghijklmnopqrstuvwxyzABCDXYZ1234567890!@#$%^&*()_+/?|}{}|":
abcdef1234567890
```

Change to above script: sanitize the `FLASHDRIVE_TOKEN` first before using it to create the `FLASHDRIVE_TOKEN_HASH`:

```bash
FLASHDRIVE_TOKEN=$(head -n1 /mnt/backup/.reset-token)
FLASHDRIVE_TOKEN=${FLASHDRIVE_TOKEN//[^a-f0-9]/}
FLASHDRIVE_TOKEN_HASH="$(echo -n "${FLASHDRIVE_TOKEN}" | sha256sum | tr -d "[:space:]-")"
```

What do you think?

Reply:
Also discussed with @NickeZ to validate that this approach is not prone to code injection.

Reply:
Why not just sha256sum the file, e.g. `FLASHDRIVE_TOKEN_HASH=sha256sum /mnt/backup/.reset-token`?

Reply:
I changed the creation of the reset token so that it does not contain anything else, not even a line break. That makes the hash of the whole file identical to the string itself, and there is no need to sanitize the reset token content.

Most elegant solution yet:
```
FLASHDRIVE_TOKEN_HASH="$(sha256sum /mnt/backup/.reset-token | cut -f 1 -d " ")"
```